### PR TITLE
*: remove TiDB only terror from parser repo

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

see more details in https://github.com/pingcap/tidb/pull/14482

### What is changed and how it works?

WIP

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/parser/715)

### CODE REVIEW

File: mysql/const.go [hunk 1]
```diff
@@ -657,21 +653,6 @@ func FormatSQLModeStr(s string) string {
...
-	for i, length := 0, len(strs); i < length; i++ {
-		mode, ok := Str2SQLMode[strs[i]]
-		if !ok && strs[i] != "" {
-			return sqlMode, newInvalidModeErr(strs[i])
-		}
-		sqlMode = sqlMode | mode
-	}
-	return sqlMode, nil
-}
-
```
Comment (kennytm) [inline mysql/const.go]: This function is used in tools and should not be removed.

Comment (lysu) [reply mysql/const.go]: thanks....I forgot tools use this package...let me take a look...

Comment (tiancaiamao) [reply mysql/const.go]: All exported functions and global variable are difficult to change

**Comment by lysu** [comment]
will take more smaller step with #734, so close this one

**Pull request closed without merging**